lib/detective/detective.dart

```diff
@@ -0,0 +1,332 @@
+    List<String> topFifeLongestWords,
+    List<String> phoneNumbersList,
+    List<String> suspiciousNumbers) {
+  investigationReport.writeAsStringSync(
+      'Number of characters in the file: $numberOfCharactersInTheFile; ',
+      mode: FileMode.append);
+  print('');
+  investigationReport.writeAsStringSync(
+      'Number of unique words: $numberOfUniqueWords; ',
+      mode: FileMode.append);
+  print('');
+
+  investigationReport.writeAsStringSync(
+      'The most frequently repeated word: $mostFrequentlyRepeatedWord; ',
+      mode: FileMode.append);
+  print('');
+  investigationReport.writeAsStringSync('Top fife longest words: ',
+      mode: FileMode.append);
+  for (int i = 0; i < topFifeLongestWords.length; i++) {
+    investigationReport.writeAsStringSync('${topFifeLongestWords[i]}, ',
+        mode: FileMode.append);
+  }
+  investigationReport.writeAsStringSync(';', mode: FileMode.append);
+  print('');
+  investigationReport.writeAsStringSync('All phone numbers: ',
+      mode: FileMode.append);
+  for (int i = 0; i < phoneNumbersList.length; i++) {
+    investigationReport.writeAsStringSync('${phoneNumbersList[i]}, ',
+        mode: FileMode.append);
+  }
+  investigationReport.writeAsStringSync(';', mode: FileMode.append);
+  print('');
+  investigationReport.writeAsStringSync('All suspicious numbers: ',
+      mode: FileMode.append);
+  for (int i = 0; i < suspiciousNumbers.length; i++) {
+    investigationReport.writeAsStringSync('${suspiciousNumbers[i]}, ',
+        mode: FileMode.append);
+  }
+  investigationReport.writeAsStringSync(';', mode: FileMode.append);
+}
+
+List<String> findSuspiciousNumbers(List<String> phoneNumbers) {
+  List<String> suspiciousNumbers = [];
+  print('All suspicious numbers: ');
+  if (phoneNumbers.isEmpty) {
+    print('There are no suspicious mobile numbers here.');
+    return [];
+  }
+  for (var phoneNumber in phoneNumbers) {
+    if (phoneNumber.startsWith('+380')) {
+      processPhoneNumber(phoneNumber, suspiciousNumbers);
+    }
+  }
+  for (int i = 0; i < suspiciousNumbers.length; i++) {
+    print(suspiciousNumbers[i]);
+  }
+  return suspiciousNumbers;
+}
+
+bool isHasSameStartAndEnd(String phoneDigits) {
+  return phoneDigits.substring(0, 3) ==
+      phoneDigits.substring(phoneDigits.length - 3);
+}
+
+bool isSymmetric(String phoneDigits) {
+  for (int i = 0; i < phoneDigits.length ~/ 2; i++) {
+    if (phoneDigits[i] != phoneDigits[phoneDigits.length - i - 1]) {
+      return false;
+    }
+  }
+  return true;
+}
+
+void processPhoneNumber(String phoneNumber, List<String> suspiciousNumbers) {
+  String phoneDigits = phoneNumber.substring(4);
+
+  if (isHasSameStartAndEnd(phoneDigits)) {
+    suspiciousNumbers.add(phoneNumber);
+    return;
+  }
+  if (isSymmetric(phoneDigits)) {
+    suspiciousNumbers.add(phoneNumber);
+  }
+}
+
+List<String> findingAllPhoneNumbers(String decipheredIntelligenceData) {
+  RegExp phoneRegExp = RegExp(r'\+\d{9,15}');
```

Comment:
don't use regular expressions it's very hard to work with it. We are training to do this by hands for now